src/modules/neo4j/neo4j-container.ts

```diff
@@ -0,0 +1,33 @@
+import { GenericContainer, Wait } from "../..";
+import { Image, Tag } from "../../repo-tag";
+import { Host } from "../../docker-client-factory";
+import { Uuid, RandomUuid } from "../../uuid";
+import { DockerClient } from "../../docker-client";
+import { BoundPorts } from "../../bound-ports";
+
+export class Neo4jContainer extends GenericContainer {
+  public static readonly defaultBoltPort = 7687;
+  public static readonly defaultHttPort = 7474;
+  public static readonly defaultUserName = "neo4j";
+  public static readonly defaultPassword = "test";
+
+  private readonly uuid: Uuid = new RandomUuid();
+  private password = Neo4jContainer.defaultPassword;
+
+  constructor(readonly image: Image = "neo4j", readonly tag: Tag = "latest", readonly host?: Host) {
+    super(image, tag);
+    this.host = host === undefined ? this.uuid.nextUuid() : host;
+    this.withName(this.host).withExposedPorts(Neo4jContainer.defaultBoltPort, Neo4jContainer.defaultHttPort);
```

Comment:
setting `host` and `withName(this.host)` isn't required - it was required for kafka because it started two containers, kafka and the zookeeper, which had to be on the same network

Reply:
Thanks for clarifying. I wasn't sure about this.